packages/client/modules/admin/containers/Graphql/GraphqlContainer.tsx

```diff
@@ -19,19 +19,50 @@ const GQL = styled('div')({
 
 type SchemaType = 'Public' | 'Private'
 
+const safeParseJSON = (str: string | null) => {
+  if (!str) return null
+  try {
+    return JSON.parse(str)
+  } catch {
+    return null
+  }
+}
+
+const persistSchemaForTab = (currentSchema: SchemaType) => {
+  // get active tab idx
+  const tabStateStr = window.localStorage.getItem('graphiql:tabState')
+  if (!tabStateStr) return
+  const parsedTabState = safeParseJSON(tabStateStr)
+  if (!parsedTabState) return
+  const activeTabIdx = parsedTabState?.activeTabIndex ?? null
+  if (activeTabIdx === null) return
```

Comment:
coded very defensively because people could edit their localStorage

Reply:
This should have been a code comment